eng/FlakyTests.targets

```diff
@@ -0,0 +1,34 @@
+<Project>
+  <Target Name="_GenerateTestPasses" DependsOnTargets="$(_GetTestsToRunTarget)">
+    <PropertyGroup>
+      <_FlakyRunAdditionalArgs>-trait "Flaky:AzDO=true"</_FlakyRunAdditionalArgs>
+      <_NonFlakyRunAdditionalArgs>-notrait "Flaky:AzDO=true"</_NonFlakyRunAdditionalArgs>
+    </PropertyGroup>
+    <ItemGroup>
+      <!-- Take the TestToRun values and update them to only run non-flaky tests -->
+      <NonFlakyRuns Include="@(TestToRun)">
+        <TestRunnerAdditionalArguments>%(TestRunnerAdditionalArguments) $(_NonFlakyRunAdditionalArgs)</TestRunnerAdditionalArguments>
+      </NonFlakyRuns>
+
+      <!-- Add new runs that do run flaky tests but ignore the exit code -->
+      <FlakyRuns Include="@(TestToRun)">
+        <TestRunnerAdditionalArguments>%(TestRunnerAdditionalArguments) $(_FlakyRunAdditionalArgs)</TestRunnerAdditionalArguments>
+        <TestRunnerIgnoreExitCode>true</TestRunnerIgnoreExitCode>
+        <ResultsHtmlPath>$([System.IO.Path]::ChangeExtension(%(ResultsHtmlPath), '.flaky.html'))</ResultsHtmlPath>
+        <ResultsStdOutPath>$([System.IO.Path]::ChangeExtension(%(ResultsStdOutPath), '.flaky.log'))</ResultsStdOutPath>
+        <ResultsXmlPath>$([System.IO.Path]::ChangeExtension(%(ResultsXmlPath), '.flaky.xml'))</ResultsXmlPath>
+      </FlakyRuns>
+
+      <!-- Replace the previous runs -->
+      <TestToRun Remove="@(TestToRun)" />
+
+
+      <!-- TODO: Replace with below when https://github.com/dotnet/arcade/issues/2182 is in -->
+      <!-- <TestToRun Include="@(NonFlakyRuns);@(FlakyRuns)" /> -->
+      <TestToRun Include="@(NonFlakyRuns)" />
```

Comment:
Note that we **do not run flaky tests at all** right now because of https://github.com/dotnet/arcade/issues/2182